src/services/utilities.ts

```diff
@@ -1303,14 +1303,7 @@ namespace ts {
     export function getScriptKind(fileName: string, host?: LanguageServiceHost): ScriptKind {
         // First check to see if the script kind was specified by the host. Chances are the host
         // may override the default script kind for the file extension.
-        let scriptKind: ScriptKind;
-        if (host && host.getScriptKind) {
-            scriptKind = host.getScriptKind(fileName);
-        }
-        if (!scriptKind) {
-            scriptKind = getScriptKindFromFileName(fileName);

```

Comment:
There's some complex logic here, but I think what I did is equivalent. In both cases:
* If `host.getScriptKind(fileName)` is defined, uses that.
* Else, uses `getScriptKindFromFileName(fileName)`.
* If that fails, uses `ScriptKind.TS`.
`ensureScriptKind` itself handles falling back to `getScriptKindFromFileName`, so there's no need to do that here.